ext/storage_sources/s3_store/s3_connection.h

```diff
@@ -15,11 +15,28 @@
 class S3Connection {
     public:
     explicit S3Connection(const Aws::S3Crt::ClientConfiguration &config);
-    bool ListBuckets(std::vector<std::string> &buckets) const;
-    bool ListObjects(const std::string &bucketName, std::vector<std::string> &objects) const;
-    bool PutObject(const std::string &bucketName, const std::string &objectKey,
+    int ListBuckets(std::vector<std::string> &buckets) const;
+
+    /*!
+     * Return a vector of object names from an S3 bucket.
+     *
+     * @param bucketName the name of the bucket.
+     * @param prefix only objects with names matching the prefix are returned.
+     * @param objects the vector of objects retrieved from S3 bucket.
+     * @param countp the number of entries returned.
+     * @param nPerIter the number of objects to access per iteration of the AWS request, from 1 to
```

Comment:
Do we handle the case if we give more than `1000`? or `0`?

Reply:
This will follow how AWS handles values of more than `1000`, or `0`, in SetMaxKey(). If given a value of more than `1000`, it will still access 1000 objects per AWS request, and if given a value of `0`, it will get 0 objects and no continuation token (ListObjects will not add any objects to the vector, before returning).

Reply:
Let's fail the function if the arguments are not in the expected range.